### PR TITLE
Better support for displaying lcd graphics

### PR DESCRIPTION
**Pull request by KevinOConnor**
This series adds some of the backend changes discussed in issue #652.  (It does not make any icon changes, but it should make it easier to implement future graphics and animations.)  This series also adds the degrees character to temperature descriptions on 128x64 displays.

@Arksine - could you verify everything still works on your displays?

-Kevin

### CODE REVIEW

File: klippy/extras/display/display.py [hunk 1]
```diff
@@ -232,16 +212,25 @@ def screen_update_128x64(self, eventtime):
...
+        pos = x
+        for i, text in enumerate(mixed_text.split('~')):
+            if pos >= max_len:
+                break
+            if i & 1 == 0:
+                # write text
+                self.lcd_chip.write_text(pos, y, text)
+                pos += len(text)
+            else:
+                # write glyph
```
Comment (Arksine) [inline klippy/extras/display/display.py]: Hi Kevin, when testing the HD44780 I noticed that the bed temperature disappears when setting a target.  I believe it is happening here.  On line 101 self.draw_heater is called with an X position of 11.  From there, self.draw_text is called with max_len=8.  I believe this is causing the loop above to break before it draws the temperature.

UC1701 is behaving as expected though.

Comment (KevinOConnor) [reply klippy/extras/display/display.py]: Thanks.  My 2004 display is packed away right now so I didn't get a chance to test it.  The max_len check wasn't needed anyway, so I removed it from the code.  Hopefully should be fixed now.

-Kevin

**Comment by Arksine** [comment]
Yep, everything works great now.

**Comment by mcmatrix** [comment]
I'll also try to test it and will see how icons will work for custom menu screens.
I have both HD44780 and st7920 screens available.
Thank you.

**Comment by mcmatrix** [comment]
I have to change how MenuManager is writing to screen a little but it looks nice on GLCD.
Have to think a little maybe I can even do dynamic icon animation by cycler.

HD44780 seems to have a problem with existing `\x7e` (in lcd it's right arrow but in PC it's '~'
so it breaks those text lines. But the solution is not to use that raw char directly but have to use `right_arrow` placeholder.

**Pull request merged**